packages/material-ui-system/src/GlobalCss/GlobalCss.js

```diff
@@ -0,0 +1,22 @@
+import { withStyles } from '@material-ui/styles';
+import spacings from './spacings';
+import colors from './colors';
+import elevations from './elevations';
+import typography from './typography';
+import positions from './positions';
+import displays from './displays';
+
+const GlobalCss = withStyles((theme) => {
```

Comment:
continuing the discussion on https://github.com/mnajdova/material-ui/pull/3#discussion_r455377139

We are now at this point (production?):

1. style generation: 10ms (OK)
2. style injection: 180ms (hum)
3. payload: 20kB gzipped (hum)

A couple of ideas around how we can improve performance:

- make the `print` media query optional (I think that most people don't care)
- remove `!important`
- work with JSS around how we can skip some of the plugins.

A couple of things to explore:

- What's the performance implication with styled-components?
- What's the performance implication with emotion?
- What will happen when the developers ask for the support of more CSS properties? e.g. all the CSS grid and flexbox properties? How will it scale? My biggest concern is with 2. as (1. seems OK and 3. can ben edged with a pruning logic, at least with JSS, is it true with emotion or styled-components?)

Reply:
A benchmark of some of the top Alexia websites, to give an idea of how far we can go:

- nytimes.com, 210kB gzipped, 1,667ms of JS
- youtube.com 56kb gzipped, 11,440ms of JS
- amazon.com 93kb gzipped, 1,562ms of JS